### PR TITLE
Add Ruby examples for bf and uuid

### PR DESCRIPTION
**Pull request by kddnewton**
I know you don't have Ruby listed in the languages you're looking for, so feel free to close if you don't want to keep it around.

### CODE REVIEW

**Comment by eatonphil** [comment]
Sweet! Totally good with the UUID implementation but could you make the BF implementation a little more like the reference implementation?

Reiterating the README, the goal isn't idiomaticness(?) but simplicity. You can keep around the more idiomatic version too though if you'd like but the main one should be the similarly simple one (i.e. I don't think you need classes).

Also, any thoughts on a Ruby port of the AATree?

File: Brainfuck/ruby/main.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      when "."
+        STDOUT.putc(data_stack[data_pointer].chr)
+      when ","
+        data_stack[data_pointer] = STDIN.getc.ord
+      when "["
+        stack = 1
+        ending = instruction_pointer + 1
+
+        while stack > 0
+          stack += { "[" => 1, "]" => -1 }.fetch(prog[ending], 0)
```
Comment (eatonphil) [inline Brainfuck/ruby/main.rb]: I don't understand what this is doing, could we simplify this?

**Comment by kddnewton** [comment]
Yeah no problem on the BF implementation, just updated. I'll take a look at AATree at some point, ran out of time this weekend.

**Comment by eatonphil** [comment]
Thanks!

File: Brainfuck/ruby/main.rb (hunk 1 above)
Comment (kddnewton) [reply Brainfuck/ruby/main.rb]: Simplified

**Pull request merged**